### ISSUE TITLE
Rename MiAbstractModel to MiAbstractVisualizationModel ?

### ISSUE THREAD

**Issue by NicolasAnquetil** [moosetechnology/MooseIDE#1167]
There are many 'model' in MooseIDE: famix model, MiAbstractModel, models of shapes in Roassal, ...

To clarify a bit things, I propose to rename MiAbstractModel to MiAbstractVisualizationModel (or something equally significant)

**Comment by ClotildeToullec** [comment]
I agree to the name change, but I'd prefer `MiAbstractBrowserModel`, because it's not always visualizations.
Also, all subclasses should be named with the same convention (`MiSomethingBrowserModel`?)

I think we should also be careful about variable names: (`aBrowserModel` /`aMooseModel` / `aShapeModel`) and method selectors (`browserModel` /`mooseModel` / `shapeModel`) when possible.
I feel like these are the most confusing.

**Comment by NicolasAnquetil** [comment]
Yes your proposed name looks good to me
Yes all subclasses should use the same convention
Yes we should be careful with the variable names. In the MiAbstractBrowser I used `specModel` which can be changed if you prefer `browserModel`.
For `shapeModel`, I am not sure we should go into Roassal...

**Comment by ClotildeToullec** [comment]
Ok, that's a plan. 
`specModel` is ok, I proposed `browserModel` because that is the convention we chose for the classes, but either are good.
Of course we don't go into roassal itself, I meant if ever we use temps or utility methods (I don't think of any example in particular, now that you mention it).
